.github/workflows/build.yml

```diff
@@ -0,0 +1,56 @@
+---
+name: Build
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+concurrency:
+  group: build-${{ github.ref_name }}
+  cancel-in-progress: true
+
+permissions:
+  contents: write
+  id-token: write
+  issues: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node:
+          - 16
+          - 17
+          - 18
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+          cache: yarn
+      - run: yarn install --immutable
+      - run: yarn compile
+      - run: yarn lint
+      - run: yarn test
+
+  deploy:
```

Comment:
@JoseLion I think this could be a separate workflow:

```yml
# release.yml
name: Release

on:
  workflow_run:
    workflows: [Build]
    types: [completed]
    branches: [release]

jobs:
  deploy:
    ...
```